assets/pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer.js

```diff
@@ -68,17 +75,17 @@ export type State = ReduxState<{|
 |}>;
 
 export type Action =
-  | { type: 'SET_STAGE', stage: Stage }
-  | { type: 'SET_FIRST_NAME', firstName: string }
-  | { type: 'SET_LAST_NAME', lastName: string }
-  | { type: 'SET_TELEPHONE', telephone: string }
-  | { type: 'SET_COUNTRY', country: string }
-  | { type: 'SET_STATE_PROVINCE', stateProvince: string }
-  | { type: 'SET_BILLING_PERIOD', billingPeriod: DigitalBillingPeriod }
-  | { type: 'SET_PAYMENT_METHOD', paymentMethod: PaymentMethod }
-  | { type: 'SET_FORM_ERRORS', errors: FormError<FormField>[] }
-  | { type: 'SET_SUBMISSION_ERROR', error: ErrorReason }
-  | { type: 'SET_FORM_SUBMITTED', formSubmitted: boolean }
+  | { type: 'SET_DP_CHECKOUT_STAGE', stage: Stage }
```

Comment:
These could all have a separate `scope` parameter on them rather than the _DP_ infix. We do something similar here: https://github.com/guardian/support-frontend/blob/ab5d62c631c91e6e92c13884025d7963d638dfc3/assets/components/marketingConsent/marketingConsentActions.js#L16

Reply:
Do we want to do this before we know for sure if the checkout infrastructure will be unified at that level? This reducer is very DP specific at the moment and it might be a better start for print if we just copypaste it as a start rather than abstract the shared logic?

Reply:
Even if we don't it feels nicer than sticking DP in the middle of everything plus it's easier to copy and paste because you just change the scope rather than all of the names.

Reply:
i suppose my concern is that we would still be "firing" 2+ reducers with the same action and trusting the reducer itself to not do anything if it's out of scope, vs implicitly marking it in the action.

I don't feel strongly about this though, and even while writing this im coming around to think it's a better future bet to use `scope` in any case

Reply:
I agree that adding a scope param will be a better solution for this. However, I don't want to mix that up with this PR because it's already been sitting around too long and I would like to get the bug fixed.

I'll revisit this in a separate PR, so have restored the old action names for now.